internal/common/types.go

```diff
@@ -138,14 +122,9 @@ type DeviceConfig struct {
 
 // ClientInfo provides the host and port of another service in the eco-system.
 type ClientInfo struct {
+	bootstrapConfig.ClientInfo
```

Comment:
Same issue with embedding bootstrapConfig.ClientInfo. In this case the timeout is used, so you just need to create a receiver functions that returns bootstrapConfig.ClientInfo.